File: Lab Report 4/C++/Task1.cpp
#include<iostream>
#include<string.h>
using namespace std;

class Employee
{
    private:
        char* name;
        char* department;
        double salary;
        double serviceTime;
    
    public:
        Employee():name(0),department(0)
        {
            salary = 0;
            serviceTime = 0;
        }

        Employee(char n[], char dep[], double sal, double t):name(n),department(dep)
        {
            salary = sal;
            serviceTime = t;
        }

        //Deep Copy Constructor
        Employee(const Employee &e)
        {
            int length = strlen(e.name);
            name = new char[length+1];
            strcpy(name, e.name);

            length = strlen(e.department);
            department = new char[length+1];
            strcpy(department, e.department);

            salary = e.salary;
            serviceTime = e.serviceTime;
        }

        void input()
        {
            string d,n;
            
			cout<<"Enter name: ";
            cin>>n;
            name = new char[n.length() + 1];
            strcpy(name, n.c_str());
            
            cout<<"Enter department: ";
            cin>>d;
            department = new char[.length() + 1];
            strcpy(department, d.c_str());
			
			cout<<"Enter salary:";
            cin>>salary;
            cout<<"Enter time of service:";
            cin>>serviceTime;
        }
        
        void show()
        {
            cout<<"\nDetails:\nName: "<<name<<endl;
            cout<<"Enter department: "<<department<<endl;
            cout<<"Enter salary:"<<salary<<endl;
            cout<<"Enter time of service:"<<serviceTime<<endl;
        }

        ~Employee()
        {
            delete[] name, department, salary, serviceTime;
        }
};

int main()
{
    Employee e1;

    e1.input();
    e1.show();

    Employee e2 = e1;

    e2.show();

    Employee e3(e1);
    
    e3.show();
    
    return 0;
}

File: Lab Report 4/Python/Task1.py
import copy


class Employee:
    def __init__(self, name, department, salary, serviceTime):
        self.name = name
        self.department = department
        self.salary = salary
        self.serviceTime = serviceTime

    def dataIn(self):
        self.name = input("Enter name: ")
        self.department = input("Enter Department: ")
        self.salary = input("Enter Salary: ")
        self.serviceTime = input("Enter Time of Service: ")

    def show(self):
        print("\n--Details--")
        print("Name: " + self.name)
        print("Department: " + self.department)
        print("Salary: " + str(self.salary))
        print("Time of Service: " + str(self.serviceTime))


e1 = Employee("", "", 0, 0)

e1.dataIn()
e1.show()

e2 = e1
e2.show()

e3 = copy.deepcopy(e1)
e3.show()